srcts/src/bindings/input/radio.ts

```diff
@@ -101,16 +107,20 @@ class RadioInputBinding extends InputBinding {
     const $el = $(el);
     // This will replace all the options
 
-    if (hasOwnProperty(data, "options")) {
+    if (hasDefinedProperty(data, "options")) {
       // Clear existing options and add each new one
       $el.find("div.shiny-options-group").remove();
       // Backward compatibility: for HTML generated by shinybootstrap2 package
       $el.find("label.radio").remove();
       // @ts-expect-error; TODO-barret; IDK what this line is doing
+      // TODO-barret; Should this line be setting attributes instead?
+      // `data.options` is an array of `{value, label}` objects
       $el.append(data.options);
```

Comment:
What are these lines doing...?

Reply:
That is pretty weird. Have you tried this code path?

Reply:
No. I'll try this soon and see if it can even be touched

Reply:
Hmmm, I tried it and it works. Super weird.

Reply:
Do you have reprex code? And what is actually being appended?

Reply:
`data.options` is just an HTML string that is all of the choices.